File: patient-community/frontend/src/data/solutions.js
/** 治疗方案静态数据，列表页与详情页共用 */
export const solutionsList = [
  {
    id: 1,
    disease_name: '糖尿病',
    title: '综合管理方案',
    description: '包括饮食控制、运动疗法和药物治疗的综合方案，通过科学的血糖管理和生活方式调整，有效控制糖尿病进展。',
    success_rate: '85%',
    user_count: 150
  },
  {
    id: 2,
    disease_name: '高血压',
    title: '降压治疗方案',
    description: '通过药物和生活方式调整控制血压，包括合理用药、低盐饮食、适量运动等综合措施。',
    success_rate: '90%',
    user_count: 200
  },
  {
    id: 3,
    disease_name: '慢性肾病',
    title: '肾功能保护方案',
    description: '延缓肾功能衰退的综合治疗方案，包括药物治疗、饮食管理和定期监测。',
    success_rate: '75%',
    user_count: 80
  },
  {
    id: 4,
    disease_name: '冠心病',
    title: '心血管康复方案',
    description: '通过药物治疗、运动康复和心理干预，改善心血管功能，预防心血管事件。',
    success_rate: '82%',
    user_count: 120
  },
  {
    id: 5,
    disease_name: '类风湿关节炎',
    title: '关节保护方案',
    description: '综合运用药物治疗、物理治疗和功能锻炼，减轻关节炎症，保护关节功能。',
    success_rate: '78%',
    user_count: 95
  },
  {
    id: 6,
    disease_name: '哮喘',
    title: '呼吸管理方案',
    description: '通过规范化的药物治疗和环境控制，有效控制哮喘症状，提高生活质量。',
    success_rate: '88%',
    user_count: 160
  }
]

export function getSolutionById(id) {
  const numId = typeof id === 'string' ? parseInt(id, 10) : id
  return solutionsList.find(s => s.id === numId) ?? null
}

File: patient-community/backend/app/api/solutions.py
from fastapi import APIRouter, Depends, HTTPException, status, Query
from sqlalchemy.orm import Session
from typing import List, Optional
from ..database import get_db
from ..models.user import User
from ..models.community import TreatmentSolution
from ..schemas.solution import SolutionCreate, SolutionResponse
from ..utils.security import get_current_user

router = APIRouter(prefix="/solutions", tags=["治疗方案"])

@router.get("", response_model=List[SolutionResponse])
def list_solutions(
    disease: Optional[str] = Query(None, description="按疾病筛选，如：糖尿病、高血压"),
    db: Session = Depends(get_db)
):
    """用户侧：仅返回已通过审核的方案，支持按疾病筛选"""
    q = (
        db.query(TreatmentSolution)
        .filter(TreatmentSolution.status == TreatmentSolution.STATUS_APPROVED)
    )
    if disease and disease.strip():
        q = q.filter(TreatmentSolution.disease_name == disease.strip())
    items = q.order_by(TreatmentSolution.updated_at.desc()).all()
    return items

@router.get("/{solution_id}", response_model=SolutionResponse)
def get_solution(solution_id: int, db: Session = Depends(get_db)):
    """用户侧：方案详情（仅已通过）"""
    item = db.query(TreatmentSolution).filter(
        TreatmentSolution.id == solution_id,
        TreatmentSolution.status == TreatmentSolution.STATUS_APPROVED
    ).first()
    if not item:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="方案不存在或未上线")
    return item

@router.post("", response_model=SolutionResponse)
def create_solution(
    data: SolutionCreate,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db)
):
    """患者新建方案，状态为待审核"""
    solution = TreatmentSolution(
        title=data.title,
        disease_name=data.disease_name,
        category=data.category or "综合方案",
        description=data.description or "",
        success_rate=data.success_rate,
        user_count=data.user_count or 0,
        status=TreatmentSolution.STATUS_PENDING,
        created_by_id=current_user.id,
    )
    db.add(solution)
    db.commit()
    db.refresh(solution)
    return solution
